tasks/RevolutionTask.cpp

```diff
@@ -153,14 +169,13 @@ void RevolutionTask::evaluateDriveCommand()
         return;
     }
 
-    base::commands::LinearAngular6DCommand compensated_drive =
-        compensateDriveCommand(drive);
+    m_compensated_command = compensateDriveCommand(drive);
 
     string drive_command =
         m_message_parser.parseDriveRevolutionCommandMessage(m_api_version,
             m_devices_id.revolution,
             m_devices_model.revolution,
-            compensated_drive);
+            m_compensated_command);
```

Comment:
why does m_compensated_command need to be a revolution attribute? it seems better to just put it as an argument, making things less convoluted.